src/Components/Common/EntityFrameworkUtils.cs

```diff
@@ -82,4 +82,14 @@ public static void PatchServiceDescriptor<TContext>(this IHostApplicationBuilder
 
         builder.Services.Add(dbContextOptionsDescriptor);
     }
+
+    public static void EnsureDbContextNotRegistered<TContext>(this IHostApplicationBuilder builder, [CallerMemberName] string callerMemberName = "") where TContext : DbContext
+    {
+        var oldDbContextOptionsDescriptor = builder.Services.FirstOrDefault(sd => sd.ServiceType == typeof(DbContextOptions<TContext>));
+
+        if (oldDbContextOptionsDescriptor is { })
```

Comment:
(nit) I think this is more readable.

```suggestion
        if (oldDbContextOptionsDescriptor is not null)
```